src/main/java/cn/nukkit/Player.java

```diff
@@ -3546,6 +3546,10 @@ public void handleDataPacket(DataPacket packet) {
                         valid = false;
                     }
 
+                    if(loginPacket.username.startsWith(" ") || loginPacket.username.startsWith(".")) {
```

Comment:
considering using `.isBlank` instead of `.startsWith(" ")`.
Also, some languages write right to left and it's common to have a space or dot at the end of one's name in some cultures.

Reply:
I think it's better to completely redo the nickname check with regex. 
In minecraft without xbox you can use most of unicode symbols. But in xbox live and nickname required to follow some rules:
Your nickname not started with space, dot
nickname not ended with space or dot
And also you can use only Latin alphabet

Why i recommend to use xbox live like nickname check system? Because with this system plugins cannot be broken in case of weird nickname.